### PR TITLE
Fabric scope

### PR DESCRIPTION
**Pull request by tecimovic**
Issue: https://github.com/project-chip/zap/issues/609

This PR adds fabric-scope flag to commands, events and attributes, adds them to all relevant queries, etc.
The schema is changed, the zcl.xsd has a new valid field `isFabriScoped="true"`.

### CODE REVIEW

File: src-electron/db/db-mapping.js [hunk 1]
```diff
@@ -112,6 +112,7 @@ exports.map = {
         x.REPORTING_POLICY == dbEnums.reportingPolicy.suggested,
       reportingPolicy: x.REPORTING_POLICY,
       isSceneRequired: dbApi.fromDbBool(x.IS_SCENE_REQUIRED),
+      isFabricScoped: dbApi.fromDbBool(x.IS_FABRIC_SCOPED),
```
Comment (bzbarsky-apple) [inline src-electron/db/db-mapping.js:115]: Attributes do not have a concept of fabric scoping in this sense, so I'm not sure we should add this here.

What has such a concept is _structs_, and adding this for struct definitions would be useful.  Then we could get rid of the `struct_is_fabric_scoped` bit in helper-zcl.js and just condition that whole "figure out what the fabric index field and the fabric sensitive fields bits are" on the fabric-scoped boolean from the DB.

File: src-electron/db/db-mapping.js [hunk 2]
```diff
@@ -142,6 +143,7 @@ exports.map = {
       description: x.DESCRIPTION,
       side: x.SIDE,
       isOptional: dbApi.fromDbBool(x.IS_OPTIONAL),
+      isFabricScoped: dbApi.fromDbBool(x.IS_FABRIC_SCOPED),
```
Comment (bzbarsky-apple) [inline src-electron/db/db-mapping.js:146]: Events do not have a concept of fabric-scoped.  They _do_ have a concept of fabric-sensitive.  And if we do this we can get rid of the `event_is_fabric_scoped` thing in helper-zcl.js, right?  Probably as a followup....

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: src-electron/db/db-mapping.js (hunk 2 above)
Comment (tecimovic) [reply src-electron/db/db-mapping.js:146]: Ok. I'll just rename this to IS_FABRIC_SENSITIVE.

File: src-electron/db/db-mapping.js (hunk 1 above)
Comment (bzbarsky-apple) [inline src-electron/db/db-mapping.js:115]: Again: attributes cannot be fabric scoped.

**Pull request merged**